app/front/scripts/services/data-package-api/index.js

```diff
@@ -194,37 +201,64 @@ function loadDimensionValues(packageId, dimension) {
 }
 
 // `dimensions` is optional; if omitted - all dimensions will be populated
-function loadDimensionsValues(packageModel, dimensions) {
+function loadDimensionsValues(packageModel, dimensions, filters) {
   return loadConfig().then(function() {
     var apiConfig = module.exports.apiConfig;
 
     dimensions = _.isArray(dimensions) ? dimensions : packageModel.dimensions;
     if (dimensions.length == 0) {
       return packageModel;
     }
+    filters = _.extend({}, filters);
 
     var promises = _.map(dimensions, function(dimension) {
-      return downloader.getJson(apiConfig.url + '/cubes/' +
+      var cut = _.clone(filters);
+      delete cut[dimension.key];
+      cut = serializeCut(cut).join('|');
+      if (cut != '') {
+        cut = '?cut=' + encodeURIComponent(cut);
+      }
+      var isNoFilters = cut == '';
+
+      var url = apiConfig.url + '/cubes/' +
         encodeURIComponent(packageModel.id) + '/members/' +
-        encodeURIComponent(dimension.id));
+        encodeURIComponent(dimension.id) + cut;
+      return downloader.getJson(url).then(function(results) {
+        results.isNoFilters = isNoFilters;
+        results.dimension = dimension;
+        return results;
+      });
     });
 
     return Promise.all(promises).then(function(results) {
-      _.each(results, function(values, index) {
-        var dimension = dimensions[index];
-        dimension.values = _.chain(results[index].data)
-          .map(function(value) {
-            var key = value[dimension.key];
-            var label = value[dimension.valueRef];
-            if (!!key) {
-              return {
-                key: key,
-                label: (label && label != key) ? key + ' - ' + label : key
-              };
-            }
-          })
-          .filter()
-          .value();
+      _.each(results, function(result) {
+        var dimension = result.dimension;
+        if (result.isNoFilters) {
+          dimension.allValues = _.chain(result.data)
+            .map(function(value) {
+              var key = value[dimension.key];
+              var label = value[dimension.valueRef];
+              if (!!key) {
+                return {
+                  key: key,
+                  label: (label && label != key) ? key + ' - ' + label : key
+                };
+              }
+            })
+            .filter()
+            .value();
+          dimension.values = dimension.allValues;
+        } else {
+          var keys = _.map(result.data, function(value) {
+            return value[dimension.key];
+          });
+          if (_.isArray(filters[dimension.key])) {
+            [].push.apply(keys, filters[dimension.key]);
+          }
+          dimension.values = _.filter(dimension.allValues, function(value) {
+            return keys.indexOf(value.key) >= 0;
```

Comment:
It's actually a "workaround" for a case that I missed when explaining my solution in openspending/openspending/issues/1114 - selecting multiple values per filter. Suppose we have a dataset:
```
 A  | B  
----+----
 a1 | b1
----+----
 a2 | b2
```
 and filters `A {a1, a2}` and `B {b1, b2}`. If user selects `(a1, a2)` - filter B will change to `{b1, b2}`, but when user will select, say, `b1` - `(a1, a2, b1)` - due to our algorithm filter A should change to `{a1}` - it will confuse the user. So I decided keep selected filters even if they should be hidden, so user will see them and will be able to remove. Actually, it were possible to remove `dimension.allValues`, but I needed it when app loads dimensions first time and there are pre-selected filters, and decided to use it to keep code a bit simpler and clear (another version of this code I tried was full of magic and that's why I rejected it).